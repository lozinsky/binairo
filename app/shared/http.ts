export function getErrorResponse(error: unknown): unknown {
  if (error instanceof Error && error.name === 'AssertionError') {
    return new Response(error.message, { status: 400 });
  }

  return error;
}
