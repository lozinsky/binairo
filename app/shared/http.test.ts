import { expect, test } from 'vitest';

import { getErrorResponse } from './http';

class AssertionError extends Error {
  constructor(message?: string) {
    super(message);

    this.name = 'AssertionError';
  }
}

test.each([new AssertionError('Message'), new Error('Message'), 'Message', '', 0, null, undefined])(
  'returns error response',
  (value) => {
    expect(getErrorResponse(value)).toMatchSnapshot();
  },
);
