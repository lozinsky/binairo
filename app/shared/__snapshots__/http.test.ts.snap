// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`returns error response 1`] = `
Response {
  Symbol(realm): {
    "settingsObject": {},
  },
  Symbol(state): {
    "aborted": false,
    "body": {
      "length": 7,
      "source": "Message",
      "stream": ReadableStream {
        Symbol(kType): "ReadableStream",
        Symbol(kState): {
          "controller": ReadableStreamDefaultController {
            Symbol(kType): "ReadableStreamDefaultController",
            Symbol(kState): {
              "cancelAlgorithm": [Function],
              "closeRequested": false,
              "highWaterMark": 1,
              "pullAgain": false,
              "pullAlgorithm": [Function],
              "pulling": false,
              "queue": [],
              "queueTotalSize": 0,
              "sizeAlgorithm": [Function],
              "started": false,
              "stream": ReadableStream {
                Symbol(kType): "ReadableStream",
                Symbol(kState): [Circular],
                Symbol(nodejs.webstream.isClosedPromise): {
                  "promise": Promise {},
                  "reject": [Function],
                  "resolve": [Function],
                },
                Symbol(nodejs.webstream.controllerErrorFunction): [Function],
              },
            },
          },
          "disturbed": false,
          "reader": undefined,
          "state": "readable",
          "storedError": undefined,
          "stream": undefined,
          "transfer": {
            "port1": undefined,
            "port2": undefined,
            "promise": undefined,
            "writable": undefined,
          },
        },
        Symbol(nodejs.webstream.isClosedPromise): {
          "promise": Promise {},
          "reject": [Function],
          "resolve": [Function],
        },
        Symbol(nodejs.webstream.controllerErrorFunction): [Function],
      },
    },
    "cacheState": "",
    "headersList": HeadersList {
      "cookies": null,
      Symbol(headers map): Map {
        "content-type" => {
          "name": "content-type",
          "value": "text/plain;charset=UTF-8",
        },
      },
      Symbol(headers map sorted): null,
    },
    "rangeRequested": false,
    "requestIncludesCredentials": false,
    "status": 400,
    "statusText": "",
    "timingAllowPassed": false,
    "timingInfo": null,
    "type": "default",
    "urlList": [],
  },
  Symbol(headers): Headers {
    Symbol(headers list): HeadersList {
      "cookies": null,
      Symbol(headers map): Map {
        "content-type" => {
          "name": "content-type",
          "value": "text/plain;charset=UTF-8",
        },
      },
      Symbol(headers map sorted): null,
    },
    Symbol(guard): "response",
    Symbol(realm): {
      "settingsObject": {},
    },
  },
}
`;

exports[`returns error response 2`] = `[Error: Message]`;

exports[`returns error response 3`] = `"Message"`;

exports[`returns error response 4`] = `""`;

exports[`returns error response 5`] = `0`;

exports[`returns error response 6`] = `null`;

exports[`returns error response 7`] = `undefined`;
